keras_rs/src/layers/embedding/jax/distributed_embedding_test.py

```diff
@@ -458,6 +464,127 @@ def loss_fn(y_true, y_pred):
+
+        table_configs = keras_test_utils.create_random_table_configs(
+            max_vocabulary_size=64,
+            max_embedding_dim=8,
+            optimizer=keras.optimizers.SGD(learning_rate=0.1),
+            seed=10,
+        )
+        feature_configs = keras_test_utils.create_random_feature_configs(
+            table_configs=table_configs,
+            batch_size=16,
+            seed=20,
+        )
+        feature_configs_dict = {
+            feature_config.name: feature_config
+            for feature_config in feature_configs
+        }
+
+        # Create tables for generating labels.
+        seed = keras.random.SeedGenerator(40)
+        tables = {
+            table_config.name: keras.random.uniform(
+                shape=(
+                    table_config.vocabulary_size,
+                    table_config.embedding_dim,
+                ),
+                minval=-5,
+                maxval=5,
+                dtype="float32",
+                seed=seed,
+            )
+            for table_config in table_configs
+        }
+
+        # Fit and evaluate.
+        def loss_fn(y_true, y_pred):
+            return jnp.mean(jnp.square(y_true - y_pred))
+
+        embedding_layer_name = "distributed_embedding_chkpt_test"
+        layer = jax_distributed_embedding.DistributedEmbedding(
+            feature_configs_dict,
+            table_stacking=target_stacking,
+            name=embedding_layer_name,
+        )
+        model = keras.Sequential([layer])
+        model.compile(jit_compile=True, loss=loss_fn)
+
+        # Fit model to different dataset.
+        training_dataset = keras_test_utils.RandomInputSampleDataset(
+            feature_configs_dict,
+            tables,
+            ragged=ragged,
+            num_batches=100,
+            seed=42,
+            preprocessor=lambda inputs, weights: layer.preprocess(
+                inputs, weights, training=True
+            ),
+        )
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            chkpt_path = os.path.join(tmp_dir, "checkpoint")
+
+        model.fit(
+            training_dataset,
+            epochs=2,
+            steps_per_epoch=1,
+            callbacks=[
+                checkpoint_utils.JaxKeras3CheckpointCallback(
+                    model, chkpt_path, max_to_keep=1, steps_per_epoch=1
+                )
+            ],
+        )
+        # Setup a model with a zero initializer but otherwise the same
+        # feature configs to test restore. Keep the same embedding layer name to
+        # ensure the correct weights are restored.
+        feature_configs_with_zero_init = {
+            feature_config.name: dataclasses.replace(
+                feature_config,
+                table=dataclasses.replace(
+                    feature_config.table, initializer="zeros"
+                ),
+            )
+            for feature_config in feature_configs
+        }
+        layer_2 = jax_distributed_embedding.DistributedEmbedding(
```

Comment:
nitpick: rename `layer_for_restore` for consistency with the other variables below